### PR TITLE
openapi(response) to specify additional response codes

### PR DESCRIPTION
**Pull request by E-gy**
TLDR:
```rust
#[get("/{id}")]
#[openapi(response(code = "404", description = "Product not found"))]
fn product(id: String) -> Json<Product> { ... }
```

# `openapi(response)`
Allows defining additional response codes (and descriptions) for routes.

Currently the only way to do that is by implementing `openapi::ResponseEntity` yourself, with one impl for each combination of responses, and further complications if one trivially tries to return a `Result` (more on that in first comment). There's no way to do so for `dyn` responses.

Now, this is done with `response` openapi config on the route, by specifying 2 mandatory named parameters - `code` and `description` (resp "HTTP Status Code" and "description" in OpenAPI). The latter is a string, and code is either number or string [(OpenAPI allows range definitions ala `4XX` for status codes)](https://swagger.io/docs/specification/describing-responses).

Optional `schema` parameter allows to specify the type to use schema from for the response.
```rs
#[openapi(response(code = 201, description = "✨", schema = "Json<Resp<String>>"))]
```
This is _particularly_ useful for routes with `dyn` responses. Also works as a workaround for documenting errors sent via warp's reject&recover.

### CODE REVIEW

**Comment by E-gy** [comment]
Personally i think the _ideal_ way to both handle and document _errors_ is something like
```rust
#[get("/")]
async fn idx() -> result<Json<Success>, Error> {...}
```
With status codes, and schemas extracted from `Success` and `Error`.

Where `Error` is _not_ a `Rejection`. This is an important part as it has been part of, long, discussion on warp - seanmonstar/warp#451.

TLDR: The difference is semantical - `Rejection`s are where filter _doesn't accept_ the request. Errors are where filter _accepts_ the request but a _runtime_ (pre)condition fails.
A good example of latter is a missing entry in the database - the request was fine, just the entry wasn't there. There's no need to go try other routes now (which is what rejection does).

warp ~~being slow on that topic and all~~ still doesn't provide proper differentiation between the 2 ||, and because of sealed traits the only way in warp to both support errors and use Rust `?` operator is with `reject::custom` which breaks in its own ways (double body consumption, prioritization for internal errors >500, ...) ||.

Oh that also means with warp you can end up with `result<result<Json<Success>, Error>, Infallible>` (with custom, tho aided, implementations for `Error`) to gen spec for `Success` and `Error` while bypassing warp's Rejection things.

rweb is layer over warp so it can act on this topic already. Or it can wait for warp to close 451 ¯\_(ツ)_/¯

I think the question here is more of "how to do it properly?" and "how many new proc macros is it going to take?", so this probably deserves to be opened in an issue... _i just put this in a comment here because this PR opened this can of worms._

**Comment by E-gy** [comment]
Implemented `schema` too.
```rs
#[openapi(response(code = 201, description = "✨", schema = "Json<Resp<String>>"))]
```

Current implementation considers the single responses entry of scheme'd type, discards everything besides [said entry's] associated response, and replaces new description. Though it has room for improvement, it is plenty enough for `Json<Error>`.
